spring-xd-dirt/src/main/java/org/springframework/xd/dirt/modules/metadata/FileReadingMode.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.modules.metadata;
+
+import java.util.Arrays;
+
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+
+/**
+ *
+ * @author Gunnar Hillert
+ * @since 1.2
+ */
+
+public enum FileReadingMode {
+
+	REF("ref", "use-ref"),
```

Comment:
@ericbottard For validation purposes - with your approach, the error response is something like:

````
'org.springframework.xd.dirt.modules.metadata.FileReadingMode' for property 'mode'; nested exception is java.lang.IllegalStateException: Cannot convert value of type [java.lang.String] to required type [org.springframework.xd.dirt.modules.metadata.FileReadingMode] for property 'mode': no matching editors or conversion strategy found
````

Wouldn't it be nice (better UX) to also return a list of all the permissible enum options?